common/djangoapps/student/models.py

```diff
@@ -148,11 +148,80 @@ class AnonymousUserId(models.Model):
 
 def anonymous_id_for_user(user, course_id, save=True):
     """
-    Return a unique id for a (user, course) pair, suitable for inserting
+    Inputs:
+        user: User model
+        course_id: string or None
+        save:  Whether the id should be saved in an AnonymousUserId object, defaults to True
+
+    Return a unique id for a (user, course_id) pair, suitable for inserting
     into e.g. personalized survey links.
 
     If user is an `AnonymousUser`, returns `None`
+    else If this user/course_id pair already has an anonymous id in AnonymousUserId object, return that
+    else: create new anonymous_id, save it in AnonymousUserId, and return anonymous id
+    """
+
+    # .. toggle_name: ANONYMOUS_USER_ID_REVERT_TO_STABLE_HASH
+    # .. toggle_implementation: SettingToggle
+    # .. toggle_default: False
+    # .. toggle_description: Used to make sure we can quickly revert back to old behaviour in case our refractoring
+    #   of anonymous_id_for_user function does not work as intended.
+    #   Reason, our refractoring adds a database lookup on every call. We are worried this will cause preformance issues.
+    # .. toggle_use_cases: temporary
+    # .. toggle_creation_date: 2021-01-26
+    # .. toggle_target_removal_date: 2021-01-29
+    # .. toggle_tickets: https://openedx.atlassian.net/browse/ARCHBOM-1645
+    if getattr(settings, "ANONYMOUS_USER_ID_REVERT_TO_STABLE_HASH"):
+        return deprecated(anonymous_id_for_user(user, course_id, save))
```

Comment:
I don't know about this deprecated keyword, can you point to some docs? Or is this a typo?

Reply:
that is a major typo